### ISSUE TITLE
Get-MgUserMemberOf fails to output all requested properties (example - onPremisesSyncEnabled)

### ISSUE THREAD

**Issue by michevnew** [microsoftgraph/msgraph-sdk-powershell#2673]
**Describe the bug**
Looks like the Get-MgUserMemberOf cmdlet fails to properly cast the output and in effect ignores a bunch of properties returned by the API. For the sake of example, let's take a look at the onPremisesSyncEnabled property.
It does not matter if the property is returned by default or specifically requested. Same behavior is observed with advanced queries as well. Beta also has similar behavior.

The bug is NOT specific to the onPremisesSyncEnabled, multiple other properties are affected. It is also not specific to the type of object returned, even for a single group output (i.e. no directory roles, no AUs) the properties will be missing (see screenshot below)

**To Reproduce**
1) Execute 
```Get-MgUserMemberOf -UserId user@domain.com -Property Id,displayName,groupTypes,mailEnabled,securityEnabled,resourceProvisioningOptions,onPremisesSyncEnabled -Debug```
2) Observe the debug output - for any relevant group object the onPremisesSyncEnabled property is returned by the Graph call
3) the AdditionalProperties blob does NOT contain the onPremisesSyncEnabled property

**Expected behavior**
the AdditionalProperties blob does contain the onPremisesSyncEnabled property

**Debug Output**
Raw API output
```    {
      "@odata.type": "#microsoft.graph.group",
      "id": "471b526b-a084-46c0-a649-986c4e2cb89d",
      "displayName": "First group",
      "groupTypes": [
        "Unified"
      ],
      "mailEnabled": true,
      "securityEnabled": false,
      "resourceProvisioningOptions": [],
      "onPremisesSyncEnabled": null
    }
```
vs cmdlet output

```
$test[3].AdditionalProperties

Key                         Value
---                         -----
@odata.type                 #microsoft.graph.group
displayName                 First group
groupTypes                  {Unified}
mailEnabled                 True
securityEnabled             False
resourceProvisioningOptions {}

```

**Module Version**
```
ModuleType Version    PreRelease Name
---------- -------    ---------- ----
Script     2.17.0                Microsoft.Graph.Authentication
Script     2.17.0                Microsoft.Graph.Users
```

**Environment Data**
```
[11:34:43][O365]# $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.4.1
PSEdition                      Core
GitCommitId                    7.4.1
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0, 5.0, 5.1, 6.0, 7.0}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Screenshots**
![image](https://github.com/microsoftgraph/msgraph-sdk-powershell/assets/38553333/594a4c29-faef-4279-a958-51dc27d2e1a6)

**Additional context**

**Comment by timayabi2020** [comment]
Hi @michevnew apologies for the late reply to your query. What you are experiencing is by design because of the SDK's code generator's (https://github.com/Azure/autorest.powershell) way of handling response entities defined in the OpenApi file. Attached is the image of the response entity defined in the open Api file used to generate the cmdlet in question.

<img width="428" alt="Image" src="https://github.com/user-attachments/assets/df7b5a03-ee97-4c34-940c-ab7d02f08405" />

Based on the code generator's rules:
1. The``id``property is fine since it has a defined type (string) and this will be displayed in the PowerShell session.
2. AdditionalProperties is a dynamic key-value object, meaning it can hold any type of data (string, int, array, another object). Therefore, AutoREST will not automatically expand ``additionalProperties`` because it doesn't know the exact structure.

In summary, for the sdk to display any property, it should have a defined type.

In case you need any further clarifications regarding the API path (``/users/{user-id}/memberOf``) responsible for this issue, kindly raise an issue here https://developer.microsoft.com/en-us/graph/support so that the API owner can respond to it.
